walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -1,23 +1,68 @@
-// Copyright 2019 Google LLC
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     https://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+// TIME COMPLEXITY: O(E * R), where E is the number of events and R is the number of members in the request
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        if (request.getDuration() > 1440)
+            return new ArrayList<>();
+        else if (events == null || events.isEmpty())
+            return new ArrayList<>(Collections.singletonList(TimeRange.fromStartEnd(0, 1440, false)));
+        else{
+            List<List<Integer>> meetingTimeRanges = new ArrayList<>();
```

Comment:
Why are TimeRanges converted to List<Integer> instead of being processed as TimeRanges throughout?

Reply:
Right, I have no idea why I did that and I didn't catch that. Thank you for the suggestion.